src/main/java/com/moviePocket/Service/impl/UserServiceImpl.java

```diff
@@ -0,0 +1,214 @@
+            return true;
+        } else
+            return false;
+    }
+
+    public boolean deleteUser(String email, String pas) {
+        User user = findUserByEmail(email);
+        if (user != null) {
+            if (passwordEncoder.matches(pas, user.getPassword())) {
+                user.setAccountActive(false);
+                user.setEmail(user.getEmail() + " not active");
+                user.setPassword("");
+                userRepository.save(user);
+                return true;
+            }
+        }
+        return false;
+    }
+
+
+    public boolean setTokenPassword(String mail) throws MessagingException {
+        User user = userRepository.findByEmail(mail);
+        if (user != null && user.getEmailVerification()) {
+            user.setTokenLostPassword(UUID.randomUUID().toString());
+            userRepository.save(user);
+            String message = String.format(
+                    "Hello, %s! \n" +
+                            "We want to make sure it's really you. Please, visit next link: http://localhost:8080/lostpassword/reset?token=%s",
+                    user.getUsername(),
+                    user.getTokenLostPassword()
+            );
+            emailSenderService.sendMailWithAttachment(user.getEmail(), message, "Reset Password");
+            return true;
+        }
+        return false;
+    }
+
+    public boolean setTokenEmail(String oldEmail, String newEmail) throws MessagingException {
+        User user = userRepository.findByEmail(oldEmail);
+        if (user != null) {
+            user.setNewEmail(newEmail);
+            user.setNewEmailToken(UUID.randomUUID().toString());
+            userRepository.save(user);
+            String message = String.format(
+                    "Hello, %s! \n" +
+                            "We want to make sure it's really you. Please, visit next link: http://localhost:8080/user/edit/newemail/%s",
+                    user.getUsername(),
+                    user.getNewEmailToken()
+            );
+            emailSenderService.sendMailWithAttachment(user.getNewEmail(), message, "New Mail Confirmation");
+            return true;
+        }
+        return false;
+    }
+
+
+    public boolean activateNewEmail(String token) {
+        User user = userRepository.findByNewEmailToken(token);
+        if (user != null) {
+            user.setEmail(user.getNewEmail());
+            user.setNewEmail(null);
+            user.setNewEmailToken(null);
+            userRepository.save(user);
+            return true;
+        }
+        return false;
+    }
+
+    public void setNewUsername(String email, String username) {
+        try {
+            User user = userRepository.findByEmail(email);
+            if (user != null && !userRepository.existsByUsername(username)) {
+                if (username.isEmpty()) {
+                    throw new MessagingException("Username cannot be empty");
+                }
+                user.setUsername(username);
```

Comment:
you removed the check for whether the username is busy

Reply:
yeah, you right, my bad